### PR TITLE
Fix homepage link position, mediumish refactor

### PR DESCRIPTION
**Pull request by thomaswallaceg**
## Ticket

* [Homepage scrolling fix; Small refactor](https://www.notion.so/xmartlabs/Homepage-returns-to-the-top-after-clicking-links-2e2ab2643ddc42c0a3393982dc23d318?pvs=4)

## Type of change

* [x] Fix
* [ ] Story
* [ ] Chore

## Description of the change

Fixing the issue required a bit of a rework and using `localStorage`. Gatsby uses a similar method to record scroll positions in every page, but I couldn't quite get it to work and this approach seems simple enough.

The blogs list and category filter were unified into a single component, which is now rendered in both the categories page and the homepage template for simplicity. Gatsby didn't like importing the graphql query from another page, so that's still duplicated.

There's also some other small fixes, like switching the "Blog" link in the `NavBar` to `/` and removing all the "Don't use `<Link>` with external links" errors I could find.

## Screenshot/Execution
https://github.com/xmartlabs/xl-blog/assets/50030585/df7d9ff3-779f-4f9c-9e16-3b195ebbf336

### CODE REVIEW

File: src/components/blog-list/blog-list.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    allMdX: PropTypes.shape({
+      group: PropTypes.arrayOf(
+        PropTypes.shape({
+          fieldValue: PropTypes.string.isRequired,
+          totalCount: PropTypes.number.isRequired,
+        }).isRequired
+      ),
+    }),
+  }),
+}
```
Comment (lauta2912) [inline src/components/blog-list/blog-list.jsx:74]: missing new line EOF

File: src/templates/blog-list.js [hunk 2]
```diff
@@ -1,55 +1,35 @@
...
-  return (
-    <>
-      <div className={styles.container}>
-        {edges.map(({ node }) => <Card data={node} key={node.id} withCategory={true} />)}     
-      </div>
-      <Pager pageContext={pageContext}/>
-    </>
-  );
+const BlogList = (props) => {
+  return <BlogListComponent {...props} />;
```
Comment (lauta2912) [inline src/templates/blog-list.js:7]: Is this only because you cannot import the GQL query elsewhere?

File: src/helpers/local-storage.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    },
+    removeItem: (key) => {
+      try {
+        sessionStorage.removeItem(key);
+      } catch {
+        return;
+      }
+    },
+  };
+
```
Comment (lauta2912) [inline src/helpers/local-storage.js:48]: new line EOF

File: src/helpers/local-storage.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    removeItem: (key) => {
+      try {
+        localStorage.removeItem(key);
+      } catch {
+        return;
+      }
+    },
+  };
+  
+  export const ErrorSafeSessionStorage = {
```
Comment (lauta2912) [inline src/helpers/local-storage.js:25]: What do you think about moving session storage to another helper file named `session-storage.js`?

**Review by lauta2912 — APPROVED** [review]

File: src/helpers/local-storage.js (hunk 4 above)
Comment (thomaswallaceg) [reply src/helpers/local-storage.js:25]: Fair enough

File: src/templates/blog-list.js (hunk 2 above)
Comment (thomaswallaceg) [reply src/templates/blog-list.js:7]: Both the query and the component had the same issue and this helped me get around the component side of things 😬

File: gatsby-browser.js [hunk 5]
```diff
@@ -51,13 +51,16 @@ const GlobalStyles = createGlobalStyle`
...
-  return (<ThemeProvider theme={Theme}>
-    <GlobalStyles />
-    <AppContextProviderComponent>
-      <Layout>
-        {element}
-      </Layout>
-    </AppContextProviderComponent>
-  </ThemeProvider>)
-}
+export const shouldUpdateScroll = () => { console.log('testttt'); return false; }
```
Comment (lauta2912) [inline gatsby-browser.js:66]: Is this function needed? don't forget to remove the console.log

**Pull request merged**